### DISCUSSION TITLE
What prompt engines are useful?

### DISCUSSION THREAD

**Discussion by dpaleka** [pszemraj/lm-api#10] [Ideas]
Currently the default "prompt engineering" step is to add a prefix and a suffix.
What other prompt engines are useful? 

Two semi-ambitious ideas:
- JSON wrapping like Riley Goodside;
- a language model call to get a better prompt.

**Comment by pszemraj** [comment]
## idea: a text-to-text prompt engine

- summarize the [awesome chatGPT prompts](https://huggingface.co/datasets/fka/awesome-chatgpt-prompts)
- train text-to-text model of "simple" summaries to the full thing
- synthetic data generation with chatGPT/other??
- see if a FLAN-t5 is useful in this regard

**Reply by pszemraj** [reply]
From a brief 80/20 investigation [pegasus-xsum](https://huggingface.co/google/pegasus-xsum) seems to (_anecdotally_) generate summaries of the prompts dataset similar to what a human might enter as the "core" prompt. 

still might take some editing; I'll see what the results look like.

**Reply by pszemraj** [reply]
examples:

![one](https://i.imgur.com/NFmaJiN.png)
---
![one](https://i.imgur.com/Z3WPGka.png)

**Reply by pszemraj** [reply]
initial pass
[augmented_chatgpt_prompts.csv](https://github.com/pszemraj/lm-api/files/10419199/augmented_chatgpt_prompts.csv)

**Reply by pszemraj** [reply]
tried with some other models too and they might be better than I thought (model name is in the filenames). let me know what you think and we can look at expanding the dataset somehow

[augmented_chatgpt_prompts-bart-large-xsum.csv](https://github.com/pszemraj/lm-api/files/10419256/augmented_chatgpt_prompts-bart-large-xsum.csv)
[augmented_chatgpt_prompts-flan-t5-base-samsum.csv](https://github.com/pszemraj/lm-api/files/10419257/augmented_chatgpt_prompts-flan-t5-base-samsum.csv)
[augmented_chatgpt_prompts-bart-large-cnn-samsum.csv](https://github.com/pszemraj/lm-api/files/10419258/augmented_chatgpt_prompts-bart-large-cnn-samsum.csv)
[augmented_chatgpt_prompts-bart-large-xsum-samsum.csv](https://github.com/pszemraj/lm-api/files/10419259/augmented_chatgpt_prompts-bart-large-xsum-samsum.csv)
